### ISSUE TITLE
Apple M1 support

### ISSUE THREAD

**Issue by evertiro** [chef/bento#1344]
### Describe the Enhancement:
I'm very specifically looking for an Ubuntu 20.04 build that can be run on an Apple M1. That said, I have an M1 MacBook Pro and am more than happy to do the legwork if someone can point me in the right direction.

### Describe the Need:
Not everyone uses an Intel-based device. All of the individual components are supported on M1, no good reason _not_ to support it.

### Current Alternative
There's a box that does work with the M1 (`rueian/ubuntu20-m1`, specifically version 1.0.0), but it's not quite right and doesn't work the same as a bento-based box!

### Can We Help You Implement This?:
I've tried to figure out how to build this on my own, but no matter what I do, packer insists that it's building with an Intel image... even though the same image can be downloaded and booted in Parallels manually. If someone can help me figure out why that's happening, I'm happy to do the rest of the work.

**Comment by evertiro** [comment]
I've been trying to get this working, but I'm still getting the following:

```
packer build -only=parallels-iso ubuntu-20.04-arm64.json
parallels-iso: output will be in this color.

==> parallels-iso: Retrieving ISO
==> parallels-iso: Trying http://cdimage.ubuntu.com/releases/20.04/release/ubuntu-20.04.2-live-server-arm64.iso
==> parallels-iso: Trying http://cdimage.ubuntu.com/releases/20.04/release/ubuntu-20.04.2-live-server-arm64.iso?checksum=sha256%3A69460b49c6781a845ca561263af3850e107814ffb9c08a19555e819d49feef4c
==> parallels-iso: http://cdimage.ubuntu.com/releases/20.04/release/ubuntu-20.04.2-live-server-arm64.iso?checksum=sha256%3A69460b49c6781a845ca561263af3850e107814ffb9c08a19555e819d49feef4c => /Users/dgriffiths/Work/Vagrant/bento/packer_templates/ubuntu/packer_cache/a06a0437be2ba6315867ed8c431f8fb1a95c27cf.iso
==> parallels-iso: Starting HTTP server on port 8147
==> parallels-iso: Creating virtual machine...
==> parallels-iso: Applying default settings...
==> parallels-iso: Creating hard drive...
==> parallels-iso: Setting the boot order...
==> parallels-iso: Attaching ISO to the default CD/DVD ROM device...
==> parallels-iso: Starting the virtual machine...
==> parallels-iso: Error starting VM: prlctl error: Failed to start the VM: Unable to start "ubuntu-20.04-arm64" because your Mac is equipped with the Apple M1 chip that does not support Intel-based operating systems. To create a compatible virtual machine, use an ISO or VHDX image file with an ARM-based operating system.
==> parallels-iso: Unregistering virtual machine...
==> parallels-iso: Deleting output directory...
Build 'parallels-iso' errored after 15 seconds 631 milliseconds: Error starting VM: prlctl error: Failed to start the VM: Unable to start "ubuntu-20.04-arm64" because your Mac is equipped with the Apple M1 chip that does not support Intel-based operating systems. To create a compatible virtual machine, use an ISO or VHDX image file with an ARM-based operating system.

==> Wait completed after 15 seconds 632 milliseconds

==> Some builds didn't complete successfully and had errors:
--> parallels-iso: Error starting VM: prlctl error: Failed to start the VM: Unable to start "ubuntu-20.04-arm64" because your Mac is equipped with the Apple M1 chip that does not support Intel-based operating systems. To create a compatible virtual machine, use an ISO or VHDX image file with an ARM-based operating system.

==> Builds finished but no artifacts were created.
```

Note that Parallels and the downloaded image both work fine if booted manually.

**Comment by nickbouwhuis** [comment]
Seems like somewhere in the `parallels-iso` builder there's some hardcoded architecture setting. Haven't found the source code for it yet.

**Comment by jyscao** [comment]
Apologies if this is already in the works, but can we expect this updated box to be available on Vagrant Cloud soon?

**Comment by swalkinshaw** [comment]
👋  apologies for the bump, but a lot of people would be very thankful if you could update the official boxes on Vagrant Cloud. The Parallels one still hasn't been published.

After that, I believe this issue is finished?

**Comment by thbar** [comment]
The box provided by @jnoxon in https://github.com/chef/bento/pull/1374 works nicely for me (just tested), but it would be great to have official builds indeed. Is there a way we can help?

**Comment by thbar** [comment]
Sharing a bit of useful information here: when running Vagrant currently, it will run in Rosetta mode. This means that a lot of commands (`echo RUBY_PLATFORM`,  or even shelling out to `uname -m`) to figure out the actual platform will return x86 instead of arm, because of the emulation. This can make it difficult to decide how to pick the correct base image.

My current trick, which finally worked, is this one:

```ruby
  platform = `sysctl -n machdep.cpu.brand_string 2>&1`
  unless $?.success?
    puts "Failed to evaluate platform"
    abort
  end
  if platform =~ /Apple M1/
    puts "Using arm64 box"
    # waiting for official images to be published, see
    # https://github.com/chef/bento/pull/1374
    # https://github.com/chef/bento/issues/1344
    config.vm.box = "jeffnoxon/ubuntu-20.04-arm64"
  else
    puts "Using x86 box"
    config.vm.box = "bento/ubuntu-20.04"
  end
```

**Comment by Stromweld** [comment]
Recently I updated the code for hcl2. I'm working on getting aarch64 boxes working in the build test pipeline and would love any help from others. Once working my plan is to publish these to vagrant cloud.

**Comment by Stromweld** [comment]
build update of `bento/ubuntu-20.04-arm64` has been released as well as a new box for ubuntu 22.04 arm64 with parallels provider.
